src/pages/TestingPage/TestControls/useControlForm.ts

```diff
@@ -0,0 +1,156 @@
+import { useEffect, useReducer, useState } from "react";
+
+type InputData = {
```

Comment:
Put the types and function in another file, one file, multiple file... as you see appropiate